src/bundler.ts

```diff
@@ -82,11 +83,29 @@ export class Bundler {
   stripComments: boolean;
   stripExcludes: UrlString[];
 
+  private _overlayUrlLoader: InMemoryOverlayUrlLoader;
+
   constructor(options?: Options) {
     const opts = options ? options : {};
-    this.analyzer = opts.analyzer ?
-        opts.analyzer :
-        new Analyzer({urlLoader: new FSUrlLoader()});
+
+    // In order for the bundler to use a given analyzer, we'll have to fork it
+    // so we can provide our own overlayUrlLoader which falls back to the
+    // analyzer's load method.
+    //
+    // TODO(usergenic): We can't delegate to `canLoad` directly, so we're
+    // proxying via the analyzer's `canResolveUrl` method.  We need to either
+    // expose `urlLoader` publicly on the analyzer or add a `canLoad` method.
+    if (opts.analyzer) {
+      this._overlayUrlLoader = new InMemoryOverlayUrlLoader({
+        canLoad: (url: string) => opts.analyzer!.canResolveUrl(url),
```

Comment:
The not-null assertions are needed because `opts` is mutable, so just because analyzer exists here and now doesn't mean it will later. Recommend storing the analyzer in a local variable and then you can drop the `!`'s

```typescript
    if (opts.analyzer) {
      const analyzer = opts.analyzer;
      this._overlayUrlLoader = new InMemoryOverlayUrlLoader({
        canLoad: (url: string) => analyzer.canResolveUrl(url),
        load: async(url: string) => analyzer.load(url),
      });
```